.github/workflows/publish_image.yaml

```
@@ -0,0 +1,72 @@
+name: Publish Image
+
+env:
+  GCLOUD_PROJECT_NUMBER: "755795606445"
+  GCLOUD_PROJECT_NAME: "trading-prod-228918"
+  ARTIFACT_REGISTRY: "europe-west2-docker.pkg.dev"
+  DOCKER_REPOSITORY: "trading-docker-repo"
+
+on:
+  workflow_call:
+    inputs:
+      image_name:
+        required: true
+        type: string
+      build_command:
+        required: true
+        type: string
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: SET UP - Checkout Code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: SET UP - Get Commit Tag
+        run: |
+          #!/bin/sh
+          set -ox
+
+          get_tag () {
+          git describe --exact-match
+          }
+
+          get_tag
+
+          if [ $? -eq 0 ]; then
+          echo "GIT_TAG=$(get_tag)" >> $GITHUB_ENV
+          else
+          echo "The commit to be built is not tagged correctly. It requires exactly one tag for the SemVer of the build. Locally, try running git tag -a VERSION && git push origin VERSION."
+          exit 1
+          fi
+
```

The `git describe` call here doesn't need to be wrapped in a function - it only bears calling once so that the flow can pivot to either bail out because of a problem, or continue past the `if` block on the happy path:
```suggestion
          if ! GIT_TAG=$(git describe --exact-match); then
            printf "The commit to be built is not tagged correctly. \
          It requires exactly one tag for the SemVer of the build. \
          Locally, try running 'git tag -a <VERSION> && git push origin <VERSION>'.\n"
            exit 1
          fi

          echo "$GIT_TAG" >> "$GITHUB_ENV"
```

---

That `if ! <COMMAND>; then` syntax is really nice error handling! TIL - thank you so much. Have cleaned up the code accordingly :)